src/common/hooks/useShouldZeroApproveSwap.ts

```diff
@@ -1,7 +1,10 @@
 import { useDerivedSwapInfo } from 'legacy/state/swap/hooks'
 import { useShouldZeroApprove } from './useShouldZeroApprove'
+import { computeSlippageAdjustedAmounts } from 'legacy/utils/prices'
 
 export function useShouldZeroApproveSwap() {
-  const { parsedAmount } = useDerivedSwapInfo()
-  return useShouldZeroApprove(parsedAmount)
+  const { v2Trade, allowedSlippage } = useDerivedSwapInfo()
+
+  const slippageAdjustedAmounts = computeSlippageAdjustedAmounts(v2Trade, allowedSlippage)
+  return useShouldZeroApprove(slippageAdjustedAmounts.INPUT)
```

Comment:
You don't need to compute both.

Example https://github.com/cowprotocol/cowswap/blob/57d43e2fb5e40bf1695cb6b8c7037cfdbd351640/src/legacy/components/swap/ConfirmSwapModal/index.tsx#L74

Just use:

```suggestion
  return useShouldZeroApprove(v2Trade.maximumAmountIn(allowedSlippage))
```

Reply:
Good point - updated the PR. Thank you.